lib/marcel/mime_type.rb

```diff
@@ -1,112 +1,98 @@
-require "mini_mime"
+module Marcel
+  class MimeType
+    BINARY = "application/octet-stream"
 
-class Marcel::MimeType
-  BINARY = "application/octet-stream"
+    class << self
+      def extend(type, extensions: [], parents: [], magic: nil)
+        existing = Marcel::TYPES[type] || [[], [], ""]
 
-  @ext_overrides = {}
+        extensions = (Array(extensions) + existing[0]).uniq
+        parents = (Array(parents) + existing[1]).uniq
+        comment = existing[2]
 
-  OverrideInfo = Struct.new(:content_type)
-
-  class << self
-    attr_reader :ext_overrides
-
-    def extend(type, extensions: [], parents: [], magic: nil)
-      existing = MimeMagic::TYPES[type] || [[], [], ""]
-
-      extensions = (Array(extensions) + existing[0]).uniq
-      parents = (Array(parents) + existing[1]).uniq
-      comment = existing[2]
-
-      MimeMagic.add(type, extensions: extensions, magic: magic, parents: parents, comment: comment)
-
-      info = MiniMime.lookup_by_content_type(type)
-      info ||= OverrideInfo.new(type)
-      extensions.each do |ext|
-        @ext_overrides[ext] = info
+        Magic.add(type, extensions: extensions, magic: magic, parents: parents, comment: comment)
       end
-    end
 
-    def for(pathname_or_io = nil, name: nil, extension: nil, declared_type: nil)
-      type_from_data = for_data(pathname_or_io)
-      fallback_type = for_declared_type(declared_type) || for_name(name) || for_extension(extension) || BINARY
+      def for(pathname_or_io = nil, name: nil, extension: nil, declared_type: nil)
+        type_from_data = for_data(pathname_or_io)
+        fallback_type = for_declared_type(declared_type) || for_name(name) || for_extension(extension) || BINARY
 
-      if type_from_data
-        most_specific_type type_from_data, fallback_type
-      else
-        fallback_type
+        if type_from_data
+          most_specific_type type_from_data, fallback_type
+        else
+          fallback_type
+        end
       end
-    end
 
-    private
-      def for_data(pathname_or_io)
-        if pathname_or_io
-          with_io(pathname_or_io) do |io|
-            if magic = MimeMagic.by_magic(io)
-              magic.type.downcase
+      private
+        def for_data(pathname_or_io)
+          if pathname_or_io
+            with_io(pathname_or_io) do |io|
+              if magic = Marcel::Magic.by_magic(io)
+                magic.type.downcase
```

Comment:
If `pathname_or_io` is an empty string or `whatever`, `magic` is `nil` and `magic.type` crashes.
```suggestion
                magic&.type&.downcase
```

Reply:
If `magic` is `nil`, this line isn’t reached.